src/tests/system/tests/test_pysss_nss_idmap.py

```diff
@@ -0,0 +1,345 @@
+        return "ipaNTSecurityIdentifier"
+    return None
+
+
+def user_setup(provider: GenericProvider, client: Client):
+    sid_name = get_sid_name(provider)
+    assert sid_name is not None
+
+    user = provider.user("user1").add()
+    user_sid = user.get([sid_name])[sid_name][0]
+
+    client.sssd.restart()
+
+    result = client.tools.getent.passwd(user.name)
+    assert result is not None, f"'{user.name}' missing"
+    assert result.name == user.name, f"'{user.name}' has wrong name {result.name}"
+    user_id = result.uid
+
+    return (user, user_id, user_sid)
+
+
+def group_setup(provider: GenericProvider, client: Client):
+    sid_name = get_sid_name(provider)
+    assert sid_name is not None
+
+    group = provider.group("group1").add()
+    group_sid = group.get([sid_name])[sid_name][0]
+
+    client.sssd.restart()
+
+    result = client.tools.getent.group(group.name)
+    assert result is not None, f"'{group.name}' missing"
+    assert result.name == group.name, f"'{group.name}' has wrong name {result.name}"
+    group_id = result.gid
+
+    return (group, group_id, group_sid)
+
+
+def run_pysss_nss_idmap(client: Client, command: str, input):
+    if isinstance(input, str):
+        input = f'"{input}"'
+
+    return client.host.conn.run(f"""python -c 'import pysss_nss_idmap; print(pysss_nss_idmap.{command}({input}))'""")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyDC)
+def test_user_by_name(client: Client, provider: GenericProvider):
+    """
+    :title: Lookup the SID of a user by name
+    :setup:
+        1. Add 'user1'
+        2. Start SSSD
+        3. Determine UID and SID as a reference
+    :steps:
+        1. Lookup SID by name
+        2. Lookup SID by user name
+        3. Lookup SID by group name
+    :expectedresults:
+        1. success
+        2. success
+        3. no result
+    :customerscenario: False
+    """
+
+    (user, user_id, user_sid) = user_setup(provider, client)
+
+    output = run_pysss_nss_idmap(client, "getsidbyname", user.name)
+    assert ast.literal_eval(output.stdout) == {user.name: {"sid": user_sid, "type": 1}}
+
+    output = run_pysss_nss_idmap(client, "getsidbyusername", user.name)
+    assert ast.literal_eval(output.stdout) == {user.name: {"sid": user_sid, "type": 1}}
+
+    output = run_pysss_nss_idmap(client, "getsidbygroupname", user.name)
+    assert ast.literal_eval(output.stdout) == {}
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyDC)
+def test_user_by_id(client: Client, provider: GenericProvider):
+    """
+    :title: Lookup the SID of a user by ID
+    :setup:
+        1. Add 'user1'
+        2. Start SSSD
+        3. Determine UID and SID as a reference
+    :steps:
+        1. Lookup SID by ID
+        2. Lookup SID by UID
+        3. Lookup SID by GID
+    :expectedresults:
+        1. success
+        2. success
+        3. no result
+    :customerscenario: False
+    """
+
```

Comment:
It's mostly about conformity, so if we extended the framework, we could keep user/group creation as part of the test.